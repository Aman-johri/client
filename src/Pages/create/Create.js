import "./create.css";
import React from "react";
import axios from "axios";
import { connect } from "react-redux";
import PropTypes from 'prop-types';

class Create extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      title: "",
      desc: "",
      categories: "",
      isButtonDisabled : true
    };
    this.handleTextChange = this.handleTextChange.bind(this);
  }
  
  handleTextChange(e){
    this.setState({ [e.target.name] : e.target.value});
    // if((this.state.title && this.state.desc && this.state.categories)){
    //   this.setState({
    //     isButtonDisabled:false
    //   })
    // }
    // else{
    //   this.setState({
    //     isButtonDisabled:true
    //   })
    // }
  }

  handleClick = () => {
    const post = {
      title: this.state.title,
      desc: this.state.desc,
      categories:this.state.categories
    };
    this.props.createData(post);
    window.location.href = "/Home";
    window.alert("Congratulation! your post has been published...");
  }
  render(){

    const {title,desc,categories,isButtonDisabled} = this.state;
    return (
      <div className="write">
        <img
          className="writeImg"
          src="https://images.pexels.com/photos/6685428/pexels-photo-6685428.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
          alt=""
        />
        <form className="writeForm">
          <div className="writeFormGroup">
            {/* <label htmlFor="fileInput">
              <i className="writeIcon fa fa-plus"></i>
            </label> */}
            {/* <input id="fileInput" type="file" style={{ display: "none" }} /> */}
            <input
              className="writeInput1"
              placeholder="Title..."
              type="text"
              id="title"
              name="title"
              // onKeyUp={this.handleEmpty}
              onChange={this.handleTextChange}
            />
            <input
             className="category"
             placeholder="Category..."
             type="text"
             id="category"
             name="categories"
            //  onKeyup={this.handleEmpty}
             onChange={this.handleTextChange}
             />


          </div>
          <div className="writeFormGroup">
            <textarea
              className="writeInput2 writeText"
              placeholder="Tell your story..."
              type="text"
              id="desc"
              name="desc"
              // onKeyUp={this.handleEmpty}
              onChange={this.handleTextChange}
            />
          </div>
          <h4 style={{marginLeft:150}}>*To enable Publish button provide all the fields value correctly</h4>
          <button className="writeSubmit"  onClick={(title && desc && categories)?this.handleClick:null} disabled={(title && desc && categories)?false:true} id="btn">
            Publish
          </button>
        </form>
      </div>
    );
  }

  }

Create.propTypes = {
  createData: PropTypes.func.isRequired
}

Create.defaultProps = {
  createData: () => {}
}

export default Create;