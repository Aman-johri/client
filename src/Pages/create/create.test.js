import Create from "./Create";
import {shallow,configure} from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

describe("Testing", () => {
    it("snapshot test for Create Post", () => {
        const component = shallow(<Create />);
        const tree = component.debug();
        expect(tree).toMatchSnapshot();
    });
    
    it("check State", () => {
    const component = shallow(<Create />);
    expect(component.state("title")).toEqual("");
    expect(component.state("desc")).toEqual("");
    });

    it("check for title field", () => {
        const component = shallow(<Create />);
        component.setState({ title: "Aman Johri" });
        expect(component.state("title")).toEqual("Aman Johri");
    });
    
    it("check for description", () => {
        const component = shallow(<Create />);
        component.setState({ desc: "Hello, I am Aman Johri" });
        expect(component.state("desc")).toEqual("Hello, I am Aman Johri");
    });

    it("Button not works when any field is empty", () => {
        const component = shallow(
          <Create />
        );
        const instance = component.instance();
        jest.spyOn(instance, "handleClick");
        component.setState({ title: "Aman Johri", desc: "Hello, I am Aman Johri" });
        component.find(".writeSubmit2").simulate("click");
        expect(instance.handleClick).toHaveBeenCalledTimes(0);
      });

    it("Button works when all field is filled", () => {
      const component = shallow(
        <Create/>
      );
      const instance = component.instance();
      jest.spyOn(instance, "handleClick");
      component.setState({ title: "Aman Johri", desc: "Hello, I am Aman Johri", categories:"life" });
      component.find(".writeSubmit2").simulate("click");
      expect(instance.handleClick).toBeCalledTimes(0);


    });
    
    });


