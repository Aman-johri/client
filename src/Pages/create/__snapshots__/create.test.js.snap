// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot test for Create Post 1`] = `
"<Fragment>
  <ForwardRef(Dialog) open={[undefined]} id=\\"diolog\\" Close={[Function (anonymous)]} fullWidth={true} maxWidth=\\"lg\\" fullHeight={true}>
    <ForwardRef(AppBar) sx={{...}}>
      <ForwardRef(Toolbar)>
        <ForwardRef(IconButton) edge=\\"start\\" color=\\"inherit\\" aria-label=\\"close\\">
          <Memo(ForwardRef(CloseIcon)) onClick={[Function (anonymous)]} id=\\"btn\\" />
        </ForwardRef(IconButton)>
        <ForwardRef(Typography) sx={{...}} variant=\\"h6\\" component=\\"div\\">
          Create Post
        </ForwardRef(Typography)>
        <button className=\\"writeSubmit2\\" onClick={{...}} id=\\"btn\\">
          Publish
        </button>
      </ForwardRef(Toolbar)>
    </ForwardRef(AppBar)>
    <div className=\\"write\\">
      <form className=\\"writeForm\\">
        <div className=\\"writeFormGroup\\">
          <input className=\\"writeInput1\\" placeholder=\\"Title...\\" type=\\"text\\" id=\\"title\\" name=\\"title\\" minlength=\\"10\\" onChange={[Function: bound handleTextChange]} required={true} />
          <select className=\\"category\\" id=\\"category\\" name=\\"categories\\" onChange={[Function: bound handleTextChange]} required={true}>
            <option value=\\"\\">
              Select Category
            </option>
            <option value=\\"Technology\\">
              Technology
            </option>
            <option value=\\"Business\\">
              Business
            </option>
            <option value=\\"Entertainment\\">
              Entertainment
            </option>
            <option value=\\"Sports\\">
              Sports
            </option>
            <option value=\\"Science\\">
              Science
            </option>
            <option value=\\"Health\\">
              Health
            </option>
          </select>
          <br />
        </div>
        <div className=\\"Error\\">
          <p style={{...}}>
            *Title must be atleast 10 characters long
          </p>
          <p style={{...}}>
            *Required
          </p>
        </div>
        <div className=\\"writeFormGroup\\">
          <input className=\\"writeInput21\\" placeholder=\\"Enter the URL of your image\\" type=\\"text\\" id=\\"img\\" name=\\"img\\" minlength=\\"10\\" onChange={[Function: bound handleTextChange]} required={true} />
        </div>
        <div className=\\"Error\\">
          <p style={{...}}>
            *Required
          </p>
        </div>
        <div className=\\"writeFormGroup\\">
          <textarea className=\\"writeInput2 writeText\\" placeholder=\\"Tell your story...\\" type=\\"text\\" id=\\"desc\\" name=\\"desc\\" onChange={[Function: bound handleTextChange]} required={true} />
        </div>
        <div className=\\"Error1\\">
          <p style={{...}}>
            *Desc must be atleast 10 character long
          </p>
        </div>
      </form>
    </div>
  </ForwardRef(Dialog)>
  <ForwardRef(Dialog) open={false} onClose={[Function (anonymous)]} aria-describedby=\\"alert-dialog-slide-description\\">
    <ForwardRef(DialogTitle)>
      Do you want to cancel your create Post?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText) id=\\"alert-dialog-slide-description\\">
        If you cancel your create post, your post will not be saved.
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button) onClick={[Function (anonymous)]}>
        Disagree
      </ForwardRef(Button)>
      <ForwardRef(Button) onClick={[Function (anonymous)]}>
        Agree
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</Fragment>"
`;
